src/refactorings/add-braces-to-if-statement/add-braces-to-if-statement.ts

```diff
@@ -0,0 +1,59 @@
+import { Editor, Code, ErrorReason } from "../../editor/editor";
+import { Selection } from "../../editor/selection";
+import * as t from "../../ast";
+
+export { addBracesToIfStatement, hasIfStatementToAddBraces };
+
+async function addBracesToIfStatement(
+  code: Code,
+  selection: Selection,
+  editor: Editor
+) {
+  const updatedCode = updateCode(t.parse(code), selection);
+
+  if (!updatedCode.hasCodeChanged) {
+    editor.showError(ErrorReason.DidNotFoundIfStatementToAddBraces);
+    return;
+  }
+
+  await editor.write(updatedCode.code);
+}
+
+function hasIfStatementToAddBraces(ast: t.AST, selection: Selection): boolean {
+  let result = false;
+
+  t.traverseAST(ast, {
```

Comment:
Here we can use the `createVisitor()` method to get rid of the duplicated logic.

We would replace this with 
```js
t.traverseAST(ast, createVisitor(selection, () => (result = true)));
```

Reply:
I agree.